protocols/virtualdefund/virtualdefund.go

```diff
@@ -74,10 +103,15 @@ func (o *Objective) Related() []protocols.Storable {
 func (o *Objective) clone() Objective {
 	clone := Objective{}
 	clone.Status = o.Status
-	vClone := o.V.Clone()
-	clone.V = vClone
 
-	// todo: #420 consider cloning for consensusChannels
+	clone.VFixed = o.VFixed.Clone()
+	clone.InitialOutcome = o.InitialOutcome.Clone()
+	clone.PaidToBob = big.NewInt(0).Set(o.PaidToBob)
+
+	clone.Signatures = [3]state.Signature{}
+	for i, s := range o.Signatures {
+		clone.Signatures[i] = s
```

Comment:
Does this clone the signatures?

(I'm not sure that's super important, TBH.)

Reply:
I believe this does! Since `state.Signature` is a struct (and not a pointer) when we assign it (on line 113) it makes a copy of the struct. I think!

Reply:
It does, but the `Signature` struct contains two slices; a copy of `s` would copy these slices, which would copy the _pointers_ that manage the slice's values.

I believe if you do:
```
s := state.Signature{...}
s_copy := s
s_copy.R = append(s_copy.R, 1)
```
then you would also be modifying the value of `s.R`?

Reply:
Good point! I forgot that the signature has slices of bytes.

I've implemented a `CloneSignature` function in [0bb26e6](https://github.com/statechannels/go-nitro/pull/512/commits/0bb26e6df92d586c427a451b686c439358f864cb) to properly clone the signature.

It probably doesn't matter at the moment but it seems like a good thing to fix before it bites us :)